src/UI/Implementation/Render/AbstractComponentRenderer.php

```diff
@@ -85,6 +92,32 @@ public function registerResources(ResourceRegistry $registry) {
 	}
 
 	/**
+	 * Bind the component to JavaScript.
+	 *
+	 * ATTENTION: If this returns an id, the returned id has to be included as id-attribute
+	 * into the HTML of your component.
+	 *
+	 * @param	JavaScriptBindable	$component
+	 * @return	string|null
+	 */
+	final protected function bindJavaScript(JavaScriptBindable $component) {
+		$binder = $component->getOnLoadCode();
+		if ($binder === null) {
+			return null;
```

Comment:
to avoid having to check in each renderer if an id is set or not and work some strange magic as seen in the close button.

Con: If ID is set only if $on_load_code is set, one can see directly from the DOM, which element have JS attached.